src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeRetrieverServiceTest.java

```diff
@@ -40,15 +41,18 @@ public void setUp() {
     @Test
     public void should_return_all_envelopes_successfully_for_a_given_jurisdiction() throws Exception {
         List<Envelope> envelopes = EnvelopeCreator.envelopes();
+        List<EnvelopeResponse> envelopeResponses = EnvelopeCreator.envelopeResponses();
 
         when(envelopeAccess.getMappings())
             .thenReturn(singletonList(new Mapping("testJurisdiction", "testService", "testService")));
 
         when(envelopeRepository.findByJurisdictionAndStatus("testJurisdiction", PROCESSED))
             .thenReturn(envelopes);
 
-        assertThat(envelopeRetrieverService.getProcessedEnvelopesByJurisdiction("testService"))
-            .containsOnly(envelopes.get(0));
+        List<EnvelopeResponse> retrievedResponses =
+            envelopeRetrieverService.getProcessedEnvelopesByJurisdiction("testService");
+        assertThat(retrievedResponses).hasSize(1);
+        assertThat(retrievedResponses.get(0)).isEqualToComparingFieldByFieldRecursively(envelopeResponses.get(0));
```

Comment:
Instead of getting first element could we not do 

```
 assertThat(retrievedResponses).containsOnly(EnvelopeCreator.expectedRespons());

```

Reply:
No, because we do not implement equals() and hashcode() and the 2 objects need to be compared field by field

Reply:
so you can do something similar to because get(0) is never good. What if you get mutliple objects ?

```
assertThat(retrievedResponses)
            .extracting("fileld1", "fileld2", "field3")
            .containsOnly(
                tuple("value1", "value2,value3)
             
     );
```

Reply:
@luigibk I think you can also do this:
```
assertThat(retrievedResponses)
  .usingFieldByFieldElementComparator()  
  .containsOnly(EnvelopeCreator.expectedRespons())
```

Reply:
@rkondratowicz that looks good but (from the docs): 
"Note that the comparison is not recursive, if one of the fields/properties is an Object, it will be compared to the other field/property using its equals method. "

Reply:
@rkondratowicz  Actually this should work:
`assertThat(retrievedResponses)
.usingRecursiveFieldByFieldElementComparator()
.containsOnly(element);`